core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -557,6 +557,55 @@ public int size() {
   public static final SqlReturnTypeInference NULLABLE_MOD =
           chain(DECIMAL_MOD_NULLABLE, ARG1_NULLABLE);
 
+  public static final SqlReturnTypeInference DECIMAL_TRUNCATE = opBinding -> {
+    RelDataTypeFactory typeFactory = opBinding.getTypeFactory();
+    RelDataType type1 = opBinding.getOperandType(0);
+    // passing random value for scale2
+    // since the default behaviour is to have the return type of result as the first operand
+    return typeFactory.getTypeSystem().deriveDecimalTruncateType(typeFactory, type1, 0);
+  };
```

Comment:
just return `type1` ?

Reply:
This enables to override the behaviour if needed.